### PR TITLE
Add Jetpack AI forms section to Forms landing page

### PR DESCRIPTION
**Pull request by IanRamosC**
Fixes pcsBup-QA-p2#comment-513

## Proposed changes:

* Add a new "AI Forms" section to the Forms landing page. The CTA links to a new post if you already have AI enabled, or to the pricing page otherwise.

Desktop | Mobile
--- | ---
image | image

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pcsBup-QA-p2#comment-513

## Does this pull request change what data or activity we track or use?

Yes. It adds a few Tracks events to the page

## Testing instructions:

* Spin up a new site using, and point to this branch using Jetpack Beta (or run it )
* Visit `/wp-admin/edit.php?post_type=feedback`
* Click on "View" on the top right corner, then on "Inbox" 
image

* Confirm that you're redirected to `/wp-admin/admin.php?page=jetpack-forms#/landing`
* Check the copy for typos, the design, and the CTA
  * If you have AI, the CTA will link to the post creation page with the Forms block open
  * If not, you'll be redirected to the pricing page

### CODE REVIEW

File: projects/packages/forms/src/dashboard/landing/index.js [hunk 1]
```diff
@@ -58,6 +87,53 @@ const LandingPage = () => {
...
+							<ul>
+								<li>
+									{ __(
+										'Creating a registration form for a global event? Automatically populate a country dropdown list with the AI Assistant.',
+										'jetpack-forms'
+									) }
+								</li>
+								<li>
+									{ __(
+										'Need a RSVP form for your event site? Simply ask AI Assistant to prepare a form that includes options for meal preferences, attendance status, or plus-ones.',
```
Comment (CodeyGuyDylan) [inline projects/packages/forms/src/dashboard/landing/index.js]: This has an orphan in it
image

Comment (CodeyGuyDylan) [inline projects/packages/forms/src/dashboard/landing/index.js]: This should start out as `Need an RSVP form`
```suggestion
										'Need an RSVP form for your event site? Simply ask AI Assistant to prepare a form that includes options for meal preferences, attendance status, or plus-ones.',
```

**Review by CodeyGuyDylan — CHANGES_REQUESTED** [review]
Other than some copy comment, I noticed when I have AI on the site, the CTA doesn't go anywhere or do anything 🤔 

https://github.com/Automattic/jetpack/assets/65001528/f1852ec5-12cc-4999-a467-5dea4620a7d0

**Comment by IanRamosC** [comment]
> I noticed when I have AI on the site, the CTA doesn't go anywhere or do anything 🤔

The function to redirect the user to the post page takes some time because it makes a request to the API to create a post, then edit it with the content of an empty form. Could you check if you're redirected to a different page after some time?
Also, if it still doesn't work, can you check if there's any error in the console?

Anyhoo, I can add a loading state to the button in a different PR if that's what's happening.

**Comment by CodeyGuyDylan** [comment]
> * /wp-admin/admin.php?page=jetpack-forms#/landing

I went ahead and tried again clicking it, and I waited several minutes and nothing is happening 🤔 

I am using the beta plugin btw

**Review by CodeyGuyDylan — APPROVED** [review]
Works for me now, thank you for the fixes!

**Pull request merged**